### ISSUE TITLE
Multi-layer doesnt work for v9.2

### ISSUE THREAD

**Issue by fatiihyildizz** [kisvegabor/lvgl#6751]
### LVGL version

v9.2

### What happened?

Hello, 
The multi-layer mechanism I worked on for v9.1 does not work in v9.2. It goes into flush callback for the first buffer but does not switch to the second buffer.

**Comment by kisvegabor** [comment]
Could you be more specific about he issue? What is your exact setup for the buffers/layers?

**Comment by fatiihyildizz** [comment]
Hello,
I am using NXP i.MX RT1176. There are 3 layers and 3 buffers for the display.
![image](https://github.com/user-attachments/assets/5a901972-d453-4a61-81a2-f5fe61e73f8e)
To flush the displays, I perform the operation by controlling the layers in a common flush callback.
![image](https://github.com/user-attachments/assets/2c771420-382a-41d4-9e75-1733bad5ccba)
For the v9.1, when I filled the displays separately, they were coming in the callback below in order. However, in v9.2, only the display I named "ps_disp_bg" comes from these displays. The same code works as it should in v9.1.

**Comment by kisvegabor** [comment]
A few notes:
- In DIRECT_MODE if multiple areas are updated on the screen the flush_cb will be called multiple times for the same screen. Therefore you need to switch frame buffers only on the last area of the the given display. You can use `lv_display_flush_is_last(display)` for this.
- Where do you call `lv_display_flush_ready`?
- With a debugger do you see if it's spinning in a loop somewhere when it stuck?

**Comment by kisvegabor** [comment]
Is it also resolved similarly to https://github.com/lvgl/lvgl/issues/6750 ?

**Comment by fatiihyildizz** [comment]
Hi @kisvegabor ,
The reason we used Multilayer was that flicker occurred on the screen during visual animations. We thought we could increase the speed by shrinking the areas with Multilayer. However, we saw that the flicker problem disappeared when the `lv_display_flush_is_last` function you mentioned above was checked in the callback. So we no longer need Multilayer. **_I recommend that those who experience flicker pay attention to this._**
